### ISSUE TITLE
Crash on MacOS Ventura 13.4

### ISSUE THREAD

**Issue by andrewclam1991** [git-ecosystem/git-credential-manager#1293]
### Version

2.1.2

### Operating system

macOS

### OS version or distribution

Ventura 13.4, x86_64 Intel Chip

### Git hosting provider(s)

GitLab

### (Azure DevOps only) What format is your remote URL?

None

### Can you access the remote repository directly in the browser?

Yes, I can access the repository

### Expected behavior

`git push -uf origin main` works

### Actual behavior

Got exception in

```
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'chars')
   at System.Text.UTF8Encoding.GetByteCount(String chars)
   at Avalonia.Native.Interop.Impl.__MicroComIAvnTextInputMethodProxy.SetSurroundingText(String text, Int32 anchorOffset, Int32 cursorOffset) in /_/src/Avalonia.Native/Interop.Generated.cs:line 4646
   at Avalonia.Native.AvaloniaNativeTextInputMethod.OnSurroundingTextChanged(Object sender, EventArgs e) in /_/src/Avalonia.Native/AvaloniaNativeTextInputMethod.cs:line 86
   at Avalonia.Native.AvaloniaNativeTextInputMethod.SetClient(ITextInputMethodClient client) in /_/src/Avalonia.Native/AvaloniaNativeTextInputMethod.cs:line 46
   at Avalonia.Input.TextInput.TextInputMethodManager.set_Client(ITextInputMethodClient value) in /_/src/Avalonia.Base/Input/TextInput/InputMethodManager.cs:line 60
   at Avalonia.Input.TextInput.TextInputMethodManager.TryFindAndApplyClient() in /_/src/Avalonia.Base/Input/TextInput/InputMethodManager.cs:line 129
   at Avalonia.Input.TextInput.TextInputMethodManager.SetFocusedElement(IInputElement element) in /_/src/Avalonia.Base/Input/TextInput/InputMethodManager.cs:line 110
   at Avalonia.Input.KeyboardDevice.SetFocusedElement(IInputElement element, NavigationMethod method, KeyModifiers keyModifiers) in /_/src/Avalonia.Base/Input/KeyboardDevice.cs:line 163
   at Avalonia.Input.FocusManager.SetFocusedElement(IFocusScope scope, IInputElement element, NavigationMethod method, KeyModifiers keyModifiers) in /_/src/Avalonia.Base/Input/FocusManager.cs:line 142
   at Avalonia.Input.FocusManager.Focus(IInputElement control, NavigationMethod method, KeyModifiers keyModifiers) in /_/src/Avalonia.Base/Input/FocusManager.cs:line 70
   at Avalonia.Input.InputElement.Focus() in /_/src/Avalonia.Base/Input/InputElement.cs:line 500
   at GitLab.UI.Views.CredentialsView.SetFocus()
   at GitCredentialManager.UI.Controls.DialogWindow.<>c__DisplayClass5_0.<OnDataContextChanged>b__2()
   at Avalonia.Threading.JobRunner.Job.Avalonia.Threading.JobRunner.IJob.Run() in /_/src/Avalonia.Base/Threading/JobRunner.cs:line 193
   at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority) in /_/src/Avalonia.Base/Threading/JobRunner.cs:line 38
   at Avalonia.Native.PlatformThreadingInterface.SignaledCallback.Signaled(Int32 priority, Int32 priorityContainsMeaningfulValue) in /_/src/Avalonia.Native/PlatformThreadingInterface.cs:line 39
   at Avalonia.Native.Interop.Impl.__MicroComIAvnSignaledCallbackVTable.Signaled(Void* this, Int32 priority, Int32 priorityContainsMeaningfulValue) in /_/src/Avalonia.Native/Interop.Generated.cs:line 5072
--- End of stack trace from previous location ---
   at Avalonia.Native.PlatformThreadingInterface.RunLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Native/PlatformThreadingInterface.cs:line 90
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken) in /_/src/Avalonia.Base/Threading/Dispatcher.cs:line 61
   at GitCredentialManager.UI.AvaloniaUi.<>c__DisplayClass6_1.<ShowWindowAsync>b__1(CancellationToken appCancelToken)
   at GitCredentialManager.UI.Dispatcher.DispatcherJob.Execute(CancellationToken ct)
   at GitCredentialManager.UI.Dispatcher.DispatcherJobQueue.Run()
   at GitCredentialManager.UI.Dispatcher.Run()
   at GitCredentialManager.Program.Main(String[] args)
error: /usr/local/share/gcm-core/git-credential-manager get died of signal 6
```

### Logs

```
Diagnose log at 2023-06-16T04:53:41Z

AppPath: /usr/local/bin/git-credential-manager
InstallDir: /usr/local/share/gcm-core/
Version: 2.1.2

------------
Diagnostic: Environment
Skipped: False
Success: True
Exception: None
Log:
OSType: macOS
OSVersion: 13.4
Reading environment variables... OK
 Variables:
"gcm-diagnose.log" 150L, 5394B
Diagnose log at 2023-06-16T04:53:41Z

AppPath: /usr/local/bin/git-credential-manager
InstallDir: /usr/local/share/gcm-core/
Version: 2.1.2

------------
Diagnostic: Environment
Skipped: False
Success: True
Exception: None
Log:
OSType: macOS
OSVersion: 13.4
Reading environment variables... OK
 Variables:
"gcm-diagnose.log" 150L, 5394B
Diagnose log at 2023-06-16T04:53:41Z

AppPath: /usr/local/bin/git-credential-manager
InstallDir: /usr/local/share/gcm-core/
Success: True
Exception: None
Log:
OSType: macOS
OSVersion: 13.4
Reading environment variables... OK
 Variables:
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.Cs35ZUBmrq/Listeners
TMPDIR=/var/folders/6g/5prsslhj59zgrf3fn17zshqm0000gn/T/
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
SHELL=/bin/zsh
HOME=/Users/andrewlam
TERMINAL_EMULATOR=JetBrains-JediTerm
GIT_TRACE2_PARENT_SID=70e19757-27ec-444c-af54-a8e418bfce71
__CFBundleIdentifier=com.jetbrains.intellij.ce
PWD=/Users/andrewlam/Desktop/codepipeline
TERM_SESSION_ID=f0ab7dc5-4c06-456b-aa11-a68b6ddd23b8
TERM=xterm-256color
JAVA_HOME=/Library/Java/JavaVirtualMachines/amazon-corretto-17.jdk/Contents/Home
COMMAND_MODE=unix2003
SHLVL=1
GIT_EXEC_PATH=/usr/local/Cellar/git/2.41.0/libexec/git-core
OLDPWD=/Users/andrewlam/Desktop/codepipeline
XPC_FLAGS=0x0
LC_CTYPE=UTF-8
XPC_SERVICE_NAME=0


------------
Diagnostic: File system
Skipped: False
Success: True
Exception: None
Log:
Temporary directory is '/var/folders/6g/5prsslhj59zgrf3fn17zshqm0000gn/T/'...
Checking basic file I/O...
Testing IFileSystem instance...
UserHomePath: /Users/andrewlam
UserDataDirectoryPath: /Users/andrewlam/.gcm
GetCurrentDirectory(): /Users/andrewlam/Desktop/codepipeline

------------
Diagnostic: Networking
Skipped: False
Success: True
Exception: None
Log:
Checking networking and HTTP stack...
Creating HTTP client... OK
IsNetworkAvailable: True
Sending HEAD request to http://example.com... OK
Sending HEAD request to https://example.com... OK
Acquiring free TCP port... OK
Testing local HTTP loopback connections...
Creating new HTTP listener for http://localhost:61265/... OK
Waiting for loopback connection... OK
Writing response... OK
Waiting for response data... OK
Loopback connection data OK

------------
Diagnostic: Git
Locating current repository... OK
Git repository at '/Users/andrewlam/Desktop/codepipeline/.git'
Listing all Git configuration... OK
Git configuration:
file:/usr/local/etc/gitconfig   credential.helper=osxkeychain
file:/Users/andrewlam/.gitconfig        pull.rebase=true
file:/Users/andrewlam/.gitconfig        core.editor=vim~
file:/Users/andrewlam/.gitconfig        core.excludesfile=/Users/andrewlam/.gitignore
file:/Users/andrewlam/.gitconfig        filter.lfs.clean=git-lfs clean -- %f
file:/Users/andrewlam/.gitconfig        filter.lfs.smudge=git-lfs smudge -- %f
file:/Users/andrewlam/.gitconfig        filter.lfs.process=git-lfs filter-process
file:/Users/andrewlam/.gitconfig        filter.lfs.required=true
file:/Users/andrewlam/.gitconfig        init.defaultbranch=mainline
file:/Users/andrewlam/.gitconfig        credential.gitlabauthmodes=pat
file:/Users/andrewlam/.gitconfig        credential.https://gitlab.aclam.dev.provider=gitlab
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=true
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        core.ignorecase=true
file:.git/config        core.precomposeunicode=true


------------
Diagnostic: Credential storage
Skipped: False
Success: True
Exception: None
Log:
ICredentialStore instance is of type: CredentialStore
Writing test credential... OK
Reading test credential... OK
Deleting test credential... OK

------------
Diagnostic: Microsoft authentication (AAD/MSA)
Skipped: False
Success: True
Exception: None
Log:
Diagnose log at 2023-06-16T04:53:41Z

AppPath: /usr/local/bin/git-credential-manager
InstallDir: /usr/local/share/gcm-core/
Version: 2.1.2

------------
Diagnostic: Environment
Skipped: False
Success: True
Exception: None
Log:
OSType: macOS
OSVersion: 13.4
Reading environment variables... OK
 Variables:
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.Cs35ZUBmrq/Listeners
TMPDIR=/var/folders/6g/5prsslhj59zgrf3fn17zshqm0000gn/T/
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
SHELL=/bin/zsh

```

**Comment by ldennington** [comment]
Can you please try:

1. Enabling tracing (`export GCM_TRACE=1 && export GIT_TRACE=1`)
2. Repro'ing the issue
3. Attaching the output (with any passwords, tokens, secrets, etc. redacted) here

**Comment by ldennington** [comment]
@andrewclam1991, @eljeffeg - are you still experiencing this issue? If so, would you mind sharing the requested trace output?

**Comment by ldennington** [comment]
Never mind - we (the maintainers) were able to repro this today. PR with workaround can be found [here](https://github.com/git-ecosystem/git-credential-manager/pull/1314).

**Comment by eljeffeg** [comment]
I ended up getting things working through GitHub desktop so not even sure how to retry as it's working now.

**Comment by mjcheetham** [comment]
Closing as the crashing issue has been fixed in #1314 and released in 2.2.1 onwards.
